airflow/cli/cli_parser.py

```diff
@@ -219,7 +219,16 @@ def _check(value):
 
 # backfill
 ARG_MARK_SUCCESS = Arg(
-    ("-m", "--mark-success"), help="Mark jobs as succeeded without running them", action="store_true"
+    ("-m", "--mark-success"),
+    action="store_const",
+    const='success',
+    dest='mark_as',
```

Comment:
If the definition of those arguments are considered public API, e.g. available for reuse in plugins, then this change is not backward compatible.

But as it is, it avoids having to add boilerplate code to check the arguments values post parsing. Minimizing the changes needed in `airflow/cli/commands/dag_command.py` and `airflow/cli/commands/task_command.py`.

Reply:
No, these consts/objects do not really form part of the public API (the CLI args do, but I'm not worried about breaking compat with this sort of change)